opentok-angular.js

```diff
@@ -311,3 +293,94 @@ ng.module('opentok', [])
       };
     }
   ]);
+
+  const captionSubscriberTracker = (OTSession, $rootScope) => {
+    const MAX_SUBS_ON_SCREEN = 4;
+    const CAPTIONS_TIMEOUT = 5 * 1000;
+
+    const captionBox = document.getElementById('caption-render-box');
+
+    let namesByConnectionId = {};
+
+    OTSession.session.on('signal:name', (event) => {
+      namesByConnectionId[event.from.connectionId] = event.data;
+    });
+    const getNameFromConnection = (connection) => {
+      let id = connection.creationTime.toString();
+      id = id.substring(id.length - 6, id.length - 1);
+      return `Guest${id}`;
+    };
+    const getName = (from) => {
+      if (!namesByConnectionId[from.connectionId]) {
+        namesByConnectionId[from.connectionId] = getNameFromConnection(from);
+      }
+      return namesByConnectionId[from.connectionId];
+    };
+
+    // we should remove the last element of the array if it exceeds the size
+    const captionsArray = $rootScope.captionsArray;
+    // the object should have the shape {caption, streamId, timeout, name}
+
+    // Actually adds the text to the box, maybe make the text smaller based on size?
+    const renderCaptionsArray = () => {
+      let captionString = '';
+      captionsArray.forEach((captionElm) => {
+        if (captionString === ''){
+          captionString = `${captionElm.name}: ${captionElm.captionText}`
+        } else {
+          captionString = `${captionString} \n ${captionElm.name}: ${captionElm.captionText}`
+        }
+      })
```

Comment:
Looks like there's two parts to this:
* concatenation of captions
* rendering of captions

Yes?  If so, maybe decouple the concatenation of captions with its rendering?  Also, easier to grok if you use a `map` with a `join` to concatenate, IMO.

Reply:
Done and done!